src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -423,6 +435,7 @@ private void removeAssociationsFor(LanguageServerDefinition definition) {
                 .filter(mapping -> definition.equals(mapping.getServerDefinition()))
                 .toList();
         fileAssociations.removeAll(mappingsToRemove);
+        definition.removeAssociations();
```

Comment:
This work uncovered a bug where stale mappings/associations were being retained in the actual language server definition instances. It was mostly benign because the registry is consulted first, and its mapping were being properly maintained, but it was still incorrect, so this ensures that the language server definition is properly cleared as well.